pkg/table/tables/partition.go

```diff
@@ -240,8 +240,8 @@ func initPartition(t *partitionedTable, def model.PartitionDefinition) (*partiti
 
 func newPartitionExpr(tblInfo *model.TableInfo, tp model.PartitionType, expr string, partCols []model.CIStr, defs []model.PartitionDefinition) (*PartitionExpr, error) {
 	// a partitioned table cannot rely on session context/sql modes, so use a default one!
-	ctx := mock.NewContext()
-	dbName := model.NewCIStr(ctx.GetSessionVars().CurrentDB)
+	ctx := contextstatic.NewStaticExprContext()
+	dbName := model.NewCIStr(ctx.GetEvalCtx().CurrentDB())
```

Comment:
Will this get the actual Current DB name? Or just the same default / empty string every time?

Reply:
It is always an empty string and same with the previous behavior with `mock.Context`. (Seems it's useless, but just keep logic unchanged)